content/en/platform/corda/4.8/open-source/key-concepts-node.md

```diff
@@ -70,33 +70,37 @@ are documented in [API: RPC operations](api-rpc.md).
 Internally, the node has access to a rich set of services that are used during flow execution to coordinate ledger
 updates. The key services provided are:
 
-* Information on other nodes on the network and the services they offer
-* Access to the contents of the vault and the storage service
-* Access to, and generation of, the node’s public-private keypairs
-* Information about the node itself
-* The current time, as tracked by the node
+* Information about other nodes on the network and the services they offer.
+* Access to the contents of the [vault](../../../../../en/platform/corda/4.8/open-source/key-concepts-vault.html) and the storage service.
+* Access to the node’s public-private key pairs.
+* Generation of new public-private key pairs.
+* Information about the node itself.
+* The current time, as tracked by the node.
 
 ## The CorDapp provider
 
-The CorDapp provider is where new CorDapps are installed to extend the behavior of the node.
+New CorDapps are installed using the CorDapp provider and are used to extend the behavior of the node.
 
-The node also has several CorDapps installed by default to handle common tasks such as:
+Your node comes with several default CorDapps installed, which handle common tasks such as:
 
-* Retrieving transactions and attachments from counterparties
-* Upgrading contracts
-* Broadcasting agreed ledger updates for recording by counterparties
+* Retrieving transactions and attachments from counterparties.
+* Upgrading contracts.
+* Broadcasting agreed ledger updates for recording by counterparties.
 
 ## Draining mode
 
-In order to operate a clean shutdown of a node, it is important than no flows are in-flight, meaning no checkpoints should
-be persisted. The node is able to be put in draining mode, during which:
+It is often necessary to spin down nodes. Planned maintenance is a common administrative task and nodes can be drained and decommissioned for a number of reasons, including for the investigation
+of performance and latency issues, maintenance, and necessary upgrading work.. Nodes can be decommissioned from the network, but in order to shut down a node cleanly, it is important that no node processes (or ['flows'](../../../../../en/platform/corda/4.8/open-source/key-concepts-flows.md)) are active,
+meaning no [checkpoints](../../../../../en/platform/corda/4.8/open-source/contributing-flow-internals.html#checkpoints) are persisted.
+
+Checkpoints freeze a flow and capture its current status, and they are automatically saved to the database when a flow suspends or resumes. A flow can be replayed from the last checkpoint if the node restarts, with this automatic checkpointing
+ensuring durability against crashes and restarts. It is important that all of this in-progress data is cleared in anticipation of shutting down a node. The draining mode ensures that before shutting down:
```

Comment:
```suggestion
Checkpoints ensure durability against crashes and restarts by freezing a flow and capturing its current status, which is automatically saved to the database. When the node is restarted, it replays the flow from the last checkpoint. 

The draining mode ensures that before shutting down:
```

I think this is misleading: It is important that all of this in-progress data is cleared in anticipation of shutting down a node. It implies that checkpoints are cleared...